openedx/adg/lms/applications/views.py

```diff
@@ -151,6 +151,29 @@ class ContactInformationView(LoginRequiredMixin, View):
     extended_profile_form = None
     application_form = None
 
+    def is_precondition_satisfied(self):
+        """
+        Checks if a written application is already submitted or not.
+
+        Returns:
+            bool: True if written application is not completed, False otherwise.
+        """
+        user_application_hub, _ = ApplicationHub.objects.get_or_create(user=self.request.user)
+
+        return not user_application_hub.is_written_application_completed
+
+    def handle_no_permission(self):
+        """
+        Redirects to application hub on get request or returns http 400 on post request.
+
+        Returns:
+            HttpResponse object.
+        """
+        if self.request.method == 'POST':
+            return HttpResponse(status=400)
```

Comment:
Just a suggestion. We can also use constant `HTTP_400_BAD_REQUEST`, from DRF, instead of plain 400